### PR TITLE
feat: add back-to-bottom btn

### PR DESCRIPTION
**Pull request by Jeffrey-mu**

### CODE REVIEW

**Comment by Jeffrey-mu** [comment]
屏幕录制2023-03-14 18 56 09

@ddiu8081  The current version adds data cache. It is necessary to go back to the input box at the bottom, which is similar to the official chat box of chat

File: src/components/BackTop.astro [hunk 1]
```diff
@@ -10,10 +10,23 @@
...
+      fill="currentColor"
+      d="M16 4L6 14l1.41 1.41L15 7.83V28h2V7.83l7.59 7.58L26 14L16 4z"
+    ></path></svg
+  >
+</button>
 <script>
   const backtop_btn = document.getElementById("backtop_btn") as HTMLElement;
+  const backbottom_btn = document.getElementById("backbottom_btn") as HTMLElement;
   backtop_btn.style.display = "none";
+  backbottom_btn.style.display = "block";
```
Comment (ddiu8081) [inline src/components/BackTop.astro:29]: I think the button should be hidden when the page scrolls to the bottom.

Comment (Jeffrey-mu) [reply src/components/BackTop.astro:29]: I also think it's fair to get to the bottom of the page and hide the button but I don't know how to judge that

Comment (yzh990918) [reply src/components/BackTop.astro:29]: Perhaps the `pin to top` and `pin to bottom` button should be grouped together in one area, with a vertical layout that looks better. Additionally, the `pin to bottom` button should be hidden when it is less than 100 pixels away from the bottom of the page.

Comment (Jeffrey-mu) [reply src/components/BackTop.astro:29]: Okay, I'll revise it

Comment (Limour-dev) [reply src/components/BackTop.astro:29]:
```js
    window.addEventListener("scroll", () => {
      if (window.scrollY > 100) {
        backtop_btn.style.display = "block";
      } else {
        backtop_btn.style.display = "none";
      };
      const scrollTop = document.documentElement.scrollTop;
      const clientHeight = document.documentElement.clientHeight;
      const scrollHeight = document.documentElement.scrollHeight;
      if (scrollTop + clientHeight >= scrollHeight - 100){
        backbottom_btn.style.display = "none";
      } else {
        backbottom_btn.style.display = "block";
      }
    });
```

Comment (Jeffrey-mu) [reply src/components/BackTop.astro:29]: Wow, that's great Thank you

File: src/components/BackTop.astro [hunk 2]
```diff
@@ -22,7 +35,11 @@
       backtop_btn.style.display = "none";
     }
   });
-
+   backbottom_btn.onclick = () => {
+    const scrollHeight = document.documentElement.scrollHeight
+    console.log(scrollHeight);
```
Comment (yzh990918) [inline src/components/BackTop.astro]: debug code should be removed.

Comment (Jeffrey-mu) [reply src/components/BackTop.astro]: The new version has been deleted. Do you need to merge and submit

**Review by yzh990918 — APPROVED** [review]

File: src/components/BackTop.astro [hunk 3]
```diff
@@ -10,19 +10,43 @@
...
+      fill="currentColor"
+      d="M16 4L6 14l1.41 1.41L15 7.83V28h2V7.83l7.59 7.58L26 14L16 4z"
+    ></path></svg
+  >
+</button>
 <script>
   const backtop_btn = document.getElementById("backtop_btn") as HTMLElement;
+  const backbottom_btn = document.getElementById("backbottom_btn") as HTMLElement;
   backtop_btn.style.display = "none";
+  backbottom_btn.style.display = "block";
```
Comment (yzh990918) [inline src/components/BackTop.astro:29]: Initially, it is none.

**Pull request merged**